src/Meshes/mesh.ts

```diff
@@ -1155,6 +1181,28 @@ export class Mesh extends AbstractMesh implements IGetSetVerticesData {
         return true;
     }
 
+    private _checkReadinessObserver: Nullable<Observer<Scene>>;
+
+    private _onMeshReadyObserverAdded = (observer: Observer<Mesh>) => {
+        // only notify once! then unregister the observer
+        observer.unregisterOnNextCall = true;
+        if (this._isCompletelyReady) {

```

Comment:
A mesh can be ready at some point, then its state changed and so not be ready anymore for some frames until it is ready again. With the current implementation, once it has been ready at some point it will always fire the `MeshReady` event (if someone registers for it) even if it becomes not ready.

Maybe `this._isCompletelyReady` should be reset to `false` here and the first part of the `if` removed...

[...] In fact, `this._isCompletelyReady` could be completely removed. In that case, you could do a function `executeWhenReady` instead (much like `scene.executeWhenReady`) and avoid creating new properties (so not hitting the 128 max properties limit). `Mesh.ExecuteWhenReady` could maybe have some value too, that would wait for an array of meshes to be ready.

Reply:
Discussed offline :-)

I agree with the fact that the private _isCompletelyReady is not needed and have removed it, moved the other privates to internal. I now add only one extra parameter